### DISCUSSION TITLE
using with vue/composition-api vue 2 to have "refetch".

### DISCUSSION THREAD

**Discussion by fenilli** [urql-graphql/urql#1671] [Q&A] [answered]
As I was reading and testing out, it seens the bindings only work with vue 3, and if you want to use vue 2 and not use wonka, you should use toPromise, but this comes with a problem that those are not refeteched on change.

How would I go in two cases:
1. Paginated table, when changing the page or filters.
2. If I'm in another page, and add a new user, how would I refetch the table, when I go back to that page.

**Comment by kitten** [comment, marked as answer]
This is a difficult one for me to answer because I don't have much experience with Vue, specifically Vue 2, as we only looked into Vue 3's composition API because it's a good fit for `urql` and have built the bindings specifically for Vue 3 to avoid having to learn more about it than our capacity is at right now.

You're completely right; `toPromise` is for small imperative calls, but will prevent any of the update logic, from the `Client` and from a caching perspective, to work.

I think the main way forward is to make sure that we can implement a Vue Options API in the future to ensure that the migration path to Vue 3 is cleared for users on Vue 2 that'd like to use `urql`: https://github.com/FormidableLabs/urql/issues/1327

Specifically it's just so easy to implement bindings with the composition API however that we'd only want to support Vue 3 right now to avoid a split in our userbase, as we've only added these bindings when Vue 3 was already released.
I can unfortunately not quite help you without any code, or specific idea of what the API looks like in Vue 2, or what even would happen if you were to use the bindings in Vue 2 😅  Sorry!
